### ISSUE TITLE
On first run: error: self.window.controller.chat.attachment.has_context(ctx.meta) : Message "object of type 'NoneType' has no len()"

### ISSUE THREAD

**Issue by WaterSibilantFalling** [szczyglis-dev/py-gpt#77]
Installed via pip install ... exactly as per the instructions. 
Linux Isis 6.0.0-5-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.0.10-2 (2022-12-01) x86_64 GNU/Linux

This is the error

```
[event] Dispatch end: KernelEvent: kernel.input.user (379)
[event] Dispatch begin: KernelEvent: kernel.response.failed (398)
[event] Before handle: 
[chat] Output ERROR: object of type 'NoneType' has no len()
Type: TypeErrorMessage: object of type 'NoneType' has no len()
Traceback:   File "/usr/local/bin/py-gpt/src/pygpt_net/core/bridge/worker.py", line 47, in run
    self.handle_additional_context()
  File "/usr/local/bin/py-gpt/src/pygpt_net/core/bridge/worker.py", line 115, in handle_additional_context
    if not self.window.controller.chat.attachment.has_context(ctx.meta):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/bin/py-gpt/src/pygpt_net/controller/chat/attachment.py", line 166, in has_context
    return len(meta.additional_ctx) > 0
           ^^^^^^^^^^^^^^^^^^^^^^^^
```

as requested in the contributing guidelines, this is the same with log-leve as DEBUG

```
** (WebKitWebProcess:15731): WARNING **: 17:28:11.851: The GStreamer FDK AAC plugin is missing, AAC playback is unlikely to work.
[LOGGER] Changing log level to: debug
** DEBUG level enabled
[event] Dispatch begin: AppEvent: input.sent (377)
[event] Before handle: {"name": "input.sent", "data": {}, "stop": false, "internal": false}
[app] Event: input.sent
[event] After handle: {"name": "input.sent", "data": {}, "stop": true, "internal": false}
[event] Dispatch end: AppEvent: input.sent (377)
[event] Dispatch begin: Event: user.send (378)
[event] Before handle: {"name": "user.send", "data": {"value": "this is a test. Please reply only with \"hello world\""}, "ctx": null, "stop": false, "internal": false}
[event] After handle: {"name": "user.send", "data": {"value": "this is a test. Please reply only with \"hello world\""}, "ctx": null, "stop": false, "internal": false}
[event] Dispatch end: Event: user.send (378)
[event] Dispatch begin: KernelEvent: kernel.input.user (379)
[event] Before handle: 
[event] Dispatch begin: KernelEvent: kernel.state.idle (379)
[event] Before handle: {"name": "kernel.state.idle", "data": {"id": "chat"}, "stop": false, "internal": false}
[event] Dispatch end: KernelEvent: kernel.state.idle (379)
[chat] Begin.
[chat] Input prompt: this is a test. Please reply only with "hello world"
[event] Dispatch begin: Event: input.before (380)
[event] Before handle: {"name": "input.before", "data": {"value": "this is a test. Please reply only with \"hello world\"", "mode": "chat"}, "ctx": null, "stop": false, "internal": false}
[event] After handle: {"name": "input.before", "data": {"value": "this is a test. Please reply only with \"hello world\"", "mode": "chat"}, "ctx": null, "stop": false, "internal": false}
[event] Dispatch end: Event: input.before (380)
[event] Dispatch begin: KernelEvent: kernel.state.busy (381)
[event] Before handle: {"name": "kernel.state.busy", "data": {"id": "chat", "msg": "Please wait..."}, "stop": false, "internal": false}
[event] Dispatch end: KernelEvent: kernel.state.busy (381)
[event] Dispatch begin: RenderEvent: render.clear.input (0)
[event] Before handle: {"name": "render.clear.input", "data": {}, "stop": false, "internal": false}
[event] Dispatch end: RenderEvent: render.clear.input (0)
[event] Dispatch begin: KernelEvent: kernel.status (384)
[event] Before handle: {"name": "kernel.status", "data": {"status": "Please wait..."}, "stop": false, "internal": false}
[event] Dispatch end: KernelEvent: kernel.status (384)
[event] Dispatch begin: Event: user.name (385)
[event] Before handle: {"name": "user.name", "data": {"value": ""}, "ctx": null, "stop": false, "internal": false}
[event] After handle: {"name": "user.name", "data": {"value": ""}, "ctx": null, "stop": false, "internal": false}
[event] Dispatch end: Event: user.name (385)
[event] Dispatch begin: Event: ai.name (386)
[event] Before handle: {"name": "ai.name", "data": {"value": ""}, "ctx": null, "stop": false, "internal": false}
[event] After handle: {"name": "ai.name", "data": {"value": ""}, "ctx": null, "stop": false, "internal": false}
[event] Dispatch end: Event: ai.name (386)
[chat] [ctx] INPUT: {"id": null, "meta": {"id": 1, "external_id": null, "uuid": "d69e0ef5-e68a-48c2-a3dc-6b7b9016effa", "name": "New", "date": "2024-11-24", "created": 1732412721, "updated": 1732422435, "indexed": 0, "mode": "chat", "model": "gpt-4o-mini", "last_mode": "chat", "last_model": "gpt-4o-mini", "thread": null, "assistant": null, "preset": null, "run": null, "status": null, "extra": null, "initialized": false, "deleted": false, "important": false, "archived": false, "label": 0, "indexes": {}, "additional_ctx": null, "group_id": 0, "root_id": null, "parent_id": null, "owner_uuid": null}, "meta_id": null, "external_id": null, "cmds": [], "results": [], "urls": [], "images": [], "images_before": [], "files": [], "attachments": [], "reply": false, "input": "this is a test. Please reply only with \"hello world\"", "output": null, "mode": "chat", "model": "gpt-4o-mini", "thread": null, "msg_id": null, "run_id": null, "input_name": "", "output_name": "", "input_timestamp": 1732422785, "output_timestamp": 1732422785, "hidden_input": null, "hidden_output": null, "input_tokens": 0, "output_tokens": 0, "total_tokens": 0, "extra": {}, "extra_ctx": null, "current": true, "internal": false, "is_vision": false, "idx": 0, "first": false, "agent_call": false, "tool_calls": [], "index_meta": {}, "doc_ids": [], "sub_calls": 0, "sub_call": false, "sub_reply": false, "hidden": false, "live": true}
[event] Dispatch begin: Event: ctx.before (387)
[event] Before handle: {"name": "ctx.before", "data": {}, "ctx": {"id": null, "meta": {"id": 1, "external_id": null, "uuid": "d69e0ef5-1111-111-1111-6b7b9016effa", "name": "New", "date": "2024-11-24", "created": 1732412721, "updated": 1732422435, "indexed": 0, "mode": "chat", "model": "gpt-4o-mini", "last_mode": "chat", "last_model": "gpt-4o-mini", "thread": null, "assistant": null, "preset": null, "run": null, "status": null, "extra": null, "initialized": false, "deleted": false, "important": false, "archived": false, "label": 0, "indexes": {}, "additional_ctx": null, "group_id": 0, "root_id": null, "parent_id": null, "owner_uuid": null}, "meta_id": null, "external_id": null, "cmds": [], "results": [], "urls": [], "images": [], "images_before": [], "files": [], "attachments": [], "reply": false, "input": "this is a test. Please reply only with \"hello world\"", "output": null, "mode": "chat", "model": "gpt-4o-mini", "thread": null, "msg_id": null, "run_id": null, "input_name": "", "output_name": "", "input_timestamp": 1732422785, "output_timestamp": 1732422785, "hidden_input": null, "hidden_output": null, "input_tokens": 0, "output_tokens": 0, "total_tokens": 0, "extra": {}, "extra_ctx": null, "current": true, "internal": false, "is_vision": false, "idx": 0, "first": false, "agent_call": false, "tool_calls": [], "index_meta": {}, "doc_ids": [], "sub_calls": 0, "sub_call": false, "sub_reply": false, "hidden": false, "live": true}, "stop": false, "internal": false}
[event] After handle: {"name": "ctx.before", "data": {}, "ctx": {"id": null, "meta": {"id": 1, "external_id": null, "uuid": "d69e0ef5-1111-1111-1111-6b7b9016effa", "name": "New", "date": "2024-11-24", "created": 1732412721, "updated": 1732422435, "indexed": 0, "mode": "chat", "model": "gpt-4o-mini", "last_mode": "chat", "last_model": "gpt-4o-mini", "thread": null, "assistant": null, "preset": null, "run": null, "status": null, "extra": null, "initialized": false, "deleted": false, "important": false, "archived": false, "label": 0, "indexes": {}, "additional_ctx": null, "group_id": 0, "root_id": null, "parent_id": null, "owner_uuid": null}, "meta_id": null, "external_id": null, "cmds": [], "results": [], "urls": [], "images": [], "images_before": [], "files": [], "attachments": [], "reply": false, "input": "this is a test. Please reply only with \"hello world\"", "output": null, "mode": "chat", "model": "gpt-4o-mini", "thread": null, "msg_id": null, "run_id": null, "input_name": "", "output_name": "", "input_timestamp": 1732422785, "output_timestamp": 1732422785, "hidden_input": null, "hidden_output": null, "input_tokens": 0, "output_tokens": 0, "total_tokens": 0, "extra": {}, "extra_ctx": null, "current": true, "internal": false, "is_vision": false, "idx": 0, "first": false, "agent_call": false, "tool_calls": [], "index_meta": {}, "doc_ids": [], "sub_calls": 0, "sub_call": false, "sub_reply": false, "hidden": false, "live": true}, "stop": false, "internal": false}
[event] Dispatch end: Event: ctx.before (387)
[event] Dispatch begin: Event: pre.prompt (388)
[event] Before handle: {"name": "pre.prompt", "data": {"mode": "chat", "value": "You are a helpful assistant."}, "ctx": null, "stop": false, "internal": false}
[event] After handle: {"name": "pre.prompt", "data": {"mode": "chat", "value": "You are a helpful assistant."}, "ctx": null, "stop": false, "internal": false}
[event] Dispatch end: Event: pre.prompt (388)
[event] Dispatch begin: Event: post.prompt (390)
[event] Before handle: {"name": "post.prompt", "data": {"mode": "chat", "reply": false, "internal": false, "value": "You are a helpful assistant.", "is_expert": false}, "ctx": {"id": null, "meta": {"id": 1, "external_id": null, "uuid": "d69e0ef5-1111-1111-1111-6b7b9016effa", "name": "New", "date": "2024-11-24", "created": 1732412721, "updated": 1732422435, "indexed": 0, "mode": "chat", "model": "gpt-4o-mini", "last_mode": "chat", "last_model": "gpt-4o-mini", "thread": null, "assistant": null, "preset": null, "run": null, "status": null, "extra": null, "initialized": false, "deleted": false, "important": false, "archived": false, "label": 0, "indexes": {}, "additional_ctx": null, "group_id": 0, "root_id": null, "parent_id": null, "owner_uuid": null}, "meta_id": null, "external_id": null, "cmds": [], "results": [], "urls": [], "images": [], "images_before": [], "files": [], "attachments": [], "reply": false, "input": "this is a test. Please reply only with \"hello world\"", "output": null, "mode": "chat", "model": "gpt-4o-mini", "thread": null, "msg_id": null, "run_id": null, "input_name": "", "output_name": "", "input_timestamp": 1732422785, "output_timestamp": 1732422785, "hidden_input": null, "hidden_output": null, "input_tokens": 0, "output_tokens": 0, "total_tokens": 0, "extra": {}, "extra_ctx": null, "current": true, "internal": false, "is_vision": false, "idx": 0, "first": false, "agent_call": false, "tool_calls": [], "index_meta": {}, "doc_ids": [], "sub_calls": 0, "sub_call": false, "sub_reply": false, "hidden": false, "live": true}, "stop": false, "internal": false}
[event] After handle: {"name": "post.prompt", "data": {"mode": "chat", "reply": false, "internal": false, "value": "You are a helpful assistant.", "is_expert": false}, "ctx": {"id": null, "meta": {"id": 1, "external_id": null, "uuid": "d69e0ef5-1111-1111-1111-6b7b9016effa", "name": "New", "date": "2024-11-24", "created": 1732412721, "updated": 1732422435, "indexed": 0, "mode": "chat", "model": "gpt-4o-mini", "last_mode": "chat", "last_model": "gpt-4o-mini", "thread": null, "assistant": null, "preset": null, "run": null, "status": null, "extra": null, "initialized": false, "deleted": false, "important": false, "archived": false, "label": 0, "indexes": {}, "additional_ctx": null, "group_id": 0, "root_id": null, "parent_id": null, "owner_uuid": null}, "meta_id": null, "external_id": null, "cmds": [], "results": [], "urls": [], "images": [], "images_before": [], "files": [], "attachments": [], "reply": false, "input": "this is a test. Please reply only with \"hello world\"", "output": null, "mode": "chat", "model": "gpt-4o-mini", "thread": null, "msg_id": null, "run_id": null, "input_name": "", "output_name": "", "input_timestamp": 1732422785, "output_timestamp": 1732422785, "hidden_input": null, "hidden_output": null, "input_tokens": 0, "output_tokens": 0, "total_tokens": 0, "extra": {}, "extra_ctx": null, "current": true, "internal": false, "is_vision": false, "idx": 0, "first": false, "agent_call": false, "tool_calls": [], "index_meta": {}, "doc_ids": [], "sub_calls": 0, "sub_call": false, "sub_reply": false, "hidden": false, "live": true}, "stop": false, "internal": false}
[event] Dispatch end: Event: post.prompt (390)
[chat] Appending input to chat window...
[event] Dispatch begin: RenderEvent: render.begin (0)
[event] Before handle: 
[event] Dispatch end: RenderEvent: render.begin (0)
[event] Dispatch begin: RenderEvent: render.input.append (0)
[event] Before handle: 
[event] Dispatch end: RenderEvent: render.input.append (0)
[event] Dispatch begin: Event: cmd.syntax (393)
[event] Before handle: {"name": "cmd.syntax", "data": {"syntax": [], "cmd": []}, "ctx": null, "stop": false, "internal": false}
[event] After handle: {"name": "cmd.syntax", "data": {"syntax": [], "cmd": []}, "ctx": null, "stop": false, "internal": false}
[event] Dispatch end: Event: cmd.syntax (393)
[event] Dispatch begin: AppEvent: input.call (394)
[event] Before handle: {"name": "input.call", "data": {}, "stop": false, "internal": false}
[app] Event: input.call
[event] After handle: {"name": "input.call", "data": {}, "stop": true, "internal": false}
[event] Dispatch end: AppEvent: input.call (394)
[event] Dispatch begin: KernelEvent: kernel.request (395)
[event] Before handle: 
[bridge] Request...
{'assistant_id': '', 'attachments': '{}', 'ctx': '{\'id\': 4, \'meta\': {\'id\': 1, \'external_id\': None, \'uuid\': \'d69e0ef5-1111-1111-1111-6b7b9016effa\', \'name\': \'New\', \'date\': \'2024-11-24\', \'created\': 1732412721, \'updated\': 1732422435, \'indexed\': 0, \'mode\': \'chat\', \'model\': \'gpt-4o-mini\', \'last_mode\': \'chat\', \'last_model\': \'gpt-4o-mini\', \'thread\': None, \'assistant\': None, \'preset\': None, \'run\': None, \'status\': None, \'extra\': None, \'initialized\': False, \'deleted\': False, \'important\': False, \'archived\': False, \'label\': 0, \'indexes\': {}, \'additional_ctx\': None, \'group_id\': 0, \'root_id\': None, \'parent_id\': None, \'owner_uuid\': None}, \'meta_id\': None, \'external_id\': None, \'cmds\': [], \'results\': [], \'urls\': [], \'images\': [], \'images_before\': [], \'files\': [], \'attachments\': [], \'reply\': False, \'input\': \'this is a test. Please reply only with "hello world"\', \'output\': None, \'mode\': \'chat\', \'model\': \'gpt-4o-mini\', \'thread\': None, \'msg_id\': None, \'run_id\': None, \'input_name\': \'\', \'output_name\': \'\', \'input_timestamp\': 1732422785, \'output_timestamp\': 1732422785, \'hidden_input\': None, \'hidden_output\': None, \'input_tokens\': 0, \'output_tokens\': 0, \'total_tokens\': 0, \'extra\': {}, \'extra_ctx\': None, \'current\': True, \'internal\': False, \'is_vision\': False, \'idx\': 0, \'first\': False, \'agent_call\': False, \'tool_calls\': [], \'index_meta\': {}, \'doc_ids\': [], \'sub_calls\': 0, \'sub_call\': False, \'sub_reply\': False, \'hidden\': False, \'live\': True}', 'external_functions': '[]', 'file_ids': '[]', 'history': '4', 'idx': 'base', 'idx_mode': 'chat', 'max_tokens': '1024', 'mode': 'chat', 'model': "{'id': 'gpt-4o-mini', 'name': 'gpt-4o-mini', 'mode': 'chat,assistant,langchain,llama_index,vision,agent,agent_llama,expert', 'langchain': {'provider': 'openai', 'mode': ['chat'], 'args': [{'name': 'model_name', 'value': 'gpt-4o-mini', 'type': 'str'}], 'env': [{'name': 'OPENAI_API_KEY', 'value': '{api_key}'}, {'name': 'OPENAI_API_BASE', 'value': '{api_endpoint}'}]}, 'ctx': 128000, 'tokens': 4096, 'default': False, 'multimodal': '', 'langchain.provider': 'openai', 'langchain.mode': 'chat', 'langchain.args': [{'name': 'model_name', 'value': 'gpt-4o-mini', 'type': 'str'}], 'langchain.env': [{'name': 'OPENAI_API_KEY', 'value': '{api_key}'}, {'name': 'OPENAI_API_BASE', 'value': '{api_endpoint}'}], 'llama_index.provider': 'openai', 'llama_index.mode': 'chat', 'llama_index.args': [{'name': 'model', 'value': 'gpt-4o-mini', 'type': 'str'}], 'llama_index.env': [{'name': 'OPENAI_API_KEY', 'value': '{api_key}'}, {'name': 'OPENAI_API_BASE', 'value': '{api_endpoint}'}]}", 'parent_mode': 'chat', 'prompt': 'this is a test. Please reply only with "hello world"', 'stream': 'False', 'system_prompt': 'You are a helpful assistant.', 'system_prompt_raw': 'You are a helpful assistant.', 'temperature': '1.0', 'thread_id': '', 'tools_outputs': '[]'}
[event] Dispatch begin: Event: mode.before (395)
[event] Before handle: {"name": "mode.before", "data": {"value": "chat", "prompt": "this is a test. Please reply only with \"hello world\""}, "ctx": {"id": 4, "meta": {"id": 1, "external_id": null, "uuid": "d69e0ef5-1111-1111-1111-6b7b9016effa", "name": "New", "date": "2024-11-24", "created": 1732412721, "updated": 1732422435, "indexed": 0, "mode": "chat", "model": "gpt-4o-mini", "last_mode": "chat", "last_model": "gpt-4o-mini", "thread": null, "assistant": null, "preset": null, "run": null, "status": null, "extra": null, "initialized": false, "deleted": false, "important": false, "archived": false, "label": 0, "indexes": {}, "additional_ctx": null, "group_id": 0, "root_id": null, "parent_id": null, "owner_uuid": null}, "meta_id": null, "external_id": null, "cmds": [], "results": [], "urls": [], "images": [], "images_before": [], "files": [], "attachments": [], "reply": false, "input": "this is a test. Please reply only with \"hello world\"", "output": null, "mode": "chat", "model": "gpt-4o-mini", "thread": null, "msg_id": null, "run_id": null, "input_name": "", "output_name": "", "input_timestamp": 1732422785, "output_timestamp": 1732422785, "hidden_input": null, "hidden_output": null, "input_tokens": 0, "output_tokens": 0, "total_tokens": 0, "extra": {}, "extra_ctx": null, "current": true, "internal": false, "is_vision": false, "idx": 0, "first": false, "agent_call": false, "tool_calls": [], "index_meta": {}, "doc_ids": [], "sub_calls": 0, "sub_call": false, "sub_reply": false, "hidden": false, "live": true}, "stop": false, "internal": false}
[event] After handle: {"name": "mode.before", "data": {"value": "chat", "prompt": "this is a test. Please reply only with \"hello world\""}, "ctx": {"id": 4, "meta": {"id": 1, "external_id": null, "uuid": "d69e0ef5-1111-1111-1111-6b7b9016effa", "name": "New", "date": "2024-11-24", "created": 1732412721, "updated": 1732422435, "indexed": 0, "mode": "chat", "model": "gpt-4o-mini", "last_mode": "chat", "last_model": "gpt-4o-mini", "thread": null, "assistant": null, "preset": null, "run": null, "status": null, "extra": null, "initialized": false, "deleted": false, "important": false, "archived": false, "label": 0, "indexes": {}, "additional_ctx": null, "group_id": 0, "root_id": null, "parent_id": null, "owner_uuid": null}, "meta_id": null, "external_id": null, "cmds": [], "results": [], "urls": [], "images": [], "images_before": [], "files": [], "attachments": [], "reply": false, "input": "this is a test. Please reply only with \"hello world\"", "output": null, "mode": "chat", "model": "gpt-4o-mini", "thread": null, "msg_id": null, "run_id": null, "input_name": "", "output_name": "", "input_timestamp": 1732422785, "output_timestamp": 1732422785, "hidden_input": null, "hidden_output": null, "input_tokens": 0, "output_tokens": 0, "total_tokens": 0, "extra": {}, "extra_ctx": null, "current": true, "internal": false, "is_vision": false, "idx": 0, "first": false, "agent_call": false, "tool_calls": [], "index_meta": {}, "doc_ids": [], "sub_calls": 0, "sub_call": false, "sub_reply": false, "hidden": false, "live": true}, "stop": false, "internal": false}
[event] Dispatch end: Event: mode.before (395)
[bridge] After inline...
{'assistant_id': '', 'attachments': '{}', 'ctx': '{\'id\': 4, \'meta\': {\'id\': 1, \'external_id\': None, \'uuid\': \'d69e0ef5-1111-1111-1111-6b7b9016effa\', \'name\': \'New\', \'date\': \'2024-11-24\', \'created\': 1732412721, \'updated\': 1732422435, \'indexed\': 0, \'mode\': \'chat\', \'model\': \'gpt-4o-mini\', \'last_mode\': \'chat\', \'last_model\': \'gpt-4o-mini\', \'thread\': None, \'assistant\': None, \'preset\': None, \'run\': None, \'status\': None, \'extra\': None, \'initialized\': False, \'deleted\': False, \'important\': False, \'archived\': False, \'label\': 0, \'indexes\': {}, \'additional_ctx\': None, \'group_id\': 0, \'root_id\': None, \'parent_id\': None, \'owner_uuid\': None}, \'meta_id\': None, \'external_id\': None, \'cmds\': [], \'results\': [], \'urls\': [], \'images\': [], \'images_before\': [], \'files\': [], \'attachments\': [], \'reply\': False, \'input\': \'this is a test. Please reply only with "hello world"\', \'output\': None, \'mode\': \'chat\', \'model\': \'gpt-4o-mini\', \'thread\': None, \'msg_id\': None, \'run_id\': None, \'input_name\': \'\', \'output_name\': \'\', \'input_timestamp\': 1732422785, \'output_timestamp\': 1732422785, \'hidden_input\': None, \'hidden_output\': None, \'input_tokens\': 0, \'output_tokens\': 0, \'total_tokens\': 0, \'extra\': {}, \'extra_ctx\': None, \'current\': True, \'internal\': False, \'is_vision\': False, \'idx\': 0, \'first\': False, \'agent_call\': False, \'tool_calls\': [], \'index_meta\': {}, \'doc_ids\': [], \'sub_calls\': 0, \'sub_call\': False, \'sub_reply\': False, \'hidden\': False, \'live\': True}', 'external_functions': '[]', 'file_ids': '[]', 'history': '4', 'idx': 'base', 'idx_mode': 'chat', 'max_tokens': '1024', 'mode': 'chat', 'model': "{'id': 'gpt-4o-mini', 'name': 'gpt-4o-mini', 'mode': 'chat,assistant,langchain,llama_index,vision,agent,agent_llama,expert', 'langchain': {'provider': 'openai', 'mode': ['chat'], 'args': [{'name': 'model_name', 'value': 'gpt-4o-mini', 'type': 'str'}], 'env': [{'name': 'OPENAI_API_KEY', 'value': '{api_key}'}, {'name': 'OPENAI_API_BASE', 'value': '{api_endpoint}'}]}, 'ctx': 128000, 'tokens': 4096, 'default': False, 'multimodal': '', 'langchain.provider': 'openai', 'langchain.mode': 'chat', 'langchain.args': [{'name': 'model_name', 'value': 'gpt-4o-mini', 'type': 'str'}], 'langchain.env': [{'name': 'OPENAI_API_KEY', 'value': '{api_key}'}, {'name': 'OPENAI_API_BASE', 'value': '{api_endpoint}'}], 'llama_index.provider': 'openai', 'llama_index.mode': 'chat', 'llama_index.args': [{'name': 'model', 'value': 'gpt-4o-mini', 'type': 'str'}], 'llama_index.env': [{'name': 'OPENAI_API_KEY', 'value': '{api_key}'}, {'name': 'OPENAI_API_BASE', 'value': '{api_endpoint}'}]}", 'parent_mode': 'chat', 'prompt': 'this is a test. Please reply only with "hello world"', 'stream': 'False', 'system_prompt': 'You are a helpful assistant.', 'system_prompt_raw': 'You are a helpful assistant.', 'temperature': '1.0', 'thread_id': '', 'tools_outputs': '[]'}
[bridge] Starting worker (async)...
[event] Dispatch end: KernelEvent: kernel.request (395)
[bridge] Worker started.
[event] Dispatch end: KernelEvent: kernel.input.user (379)
[event] Dispatch begin: KernelEvent: kernel.response.failed (398)
[event] Before handle: 
[chat] Output ERROR: object of type 'NoneType' has no len()
Type: TypeErrorMessage: object of type 'NoneType' has no len()
Traceback:   File "/usr/local/bin/py-gpt/src/pygpt_net/core/bridge/worker.py", line 47, in run
    self.handle_additional_context()
  File "/usr/local/bin/py-gpt/src/pygpt_net/core/bridge/worker.py", line 115, in handle_additional_context
    if not self.window.controller.chat.attachment.has_context(ctx.meta):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/bin/py-gpt/src/pygpt_net/controller/chat/attachment.py", line 166, in has_context
    return len(meta.additional_ctx) > 0
           ^^^^^^^^^^^^^^^^^^^^^^^^

[event] Dispatch begin: KernelEvent: kernel.status (398)
[event] Before handle: {"name": "kernel.status", "data": {"status": "object of type 'NoneType' has no len()"}, "stop": false, "internal": false}
[event] Dispatch end: KernelEvent: kernel.status (398)
[event] Dispatch begin: AppEvent: input.error (399)
[event] Before handle: {"name": "input.error", "data": {}, "stop": false, "internal": false}
[app] Event: input.error
[event] After handle: {"name": "input.error", "data": {}, "stop": true, "internal": false}
[event] Dispatch end: AppEvent: input.error (399)
Error in sending text: object of type 'NoneType' has no len()
[event] Dispatch begin: KernelEvent: kernel.state.error (400)
[event] Before handle: {"name": "kernel.state.error", "data": {"id": "chat"}, "stop": false, "internal": false}
[event] Dispatch end: KernelEvent: kernel.state.error (400)
[event] Dispatch end: KernelEvent: kernel.response.failed (398)
```

**Comment by szczyglis-dev** [comment]
Fixed in [2.4.26](https://github.com/szczyglis-dev/py-gpt/releases/tag/v2.4.26)

**Comment by WaterSibilantFalling** [comment]
Awesome dude. Stuff like this is the magical kingdom of open source:
- As a first-time newbie I file a bug, 
- in, what, 2 days you - the maintainer has (1) fixed the bug (2) issued a fixed, updated version
- everyone is happy (a.k.a "profit")
